documentation/correctness.adoc

```diff
@@ -0,0 +1,106 @@
+= Model correctness
+
+Whether a model is _correct_ –- or _valid_, or _consistent_ –- is an important question.
+Correctness ultimately determines whether a model is useful in the context it exists in.
+Answering that question only with a simple "`yes`" or "`no`" is too simplistic: a model can be incorrect in different ways and on different levels.
+
+Not being able to resolve a reference is a different kind of failure than violating a semantic constraint.
+Having an illegal extra comma in an array somewhere in the JSON serialization chunk of a model -- e.g. `[ 0, 1, 2**,** ]` -- should not prevent that serialization chunk from being deserializable.
+
+[IMPORTANT]
+====
+We'll talk about _correctness_ from now, and consider _validity_ and _consistency_ to be equivalent concepts.
+====
+
+It makes sense to phrase the answer as a collection of _issues_, instead of giving a hard "`yes/no`" answer.
+An issue is a specific point in the model where a specific kind of incorrectness arises.
+Many (kinds of) issues can be meaningfully recovered from, and recovery can often be obvious to the point it can be automatic.
+The unresolved reference can be left in as-is, and treated as an unknown (but not a _missing_) value with special meaning (i.e., domain-specific semantics).
```

Comment:
Maybe we could state that this may be possible only for advanced consumers of nodes. Some may be able to work only with correct models (for example, most interpreters)

Reply:
I added text for that.